src/Project.php

```diff
@@ -22,6 +22,7 @@ class Project
 
     const XPATH_PROJECT_NAME = "(//component[@name='ProjectView']/panes/pane[@id='ProjectPane']/subPane/PATH/PATH_ELEMENT/option/@value)[1]";
     const XPATH_PROJECT_NAME_ALT = "(//component[@name='ProjectView']/panes/pane[@id='ProjectPane']/subPane/expand/path/item[contains(@type, ':ProjectViewProjectNode')]/@name)[1]";
+    const XPATH_PROJECT_NAME_AS = "//string[@name='app_name']";
```

Comment:
Ok I asked my teammate and his projects have a few different `app_name`. Ex: `app_name_prod`, `app_name_clientname`, `app_name_otherclientname`.

maybe grab the first `app_name` or something like that. I wonder what `android-studio` use internally to locate the name as the one in there is not the same as the one listed in the project window.

Reply:
I try to find answer in test project called "My First Application", can you try too ?

```bash
→ grep "My First Application" AndroidStudioProjects/MyFirstApplication/.* -R
AndroidStudioProjects/MyFirstApplication/.idea/workspace.xml:    <ignored path="My First Application.iws" />
AndroidStudioProjects/MyFirstApplication/.idea/workspace.xml:    <favorites_list name="My First Application" />

→ grep "My First Application" AndroidStudioProjects/MyFirstApplication/* -R
AndroidStudioProjects/MyFirstApplication/app/build/intermediates/incremental/mergeDebugResources/merged.dir/values/values.xml:    <string name="app_name">My First Application</string>
AndroidStudioProjects/MyFirstApplication/app/build/intermediates/incremental/mergeDebugResources/merger.xml:    </string-array><string name="pref_default_display_name">John Smith</string><string name="app_name">My First Application</string><string name="pref_description_social_recommendations">Recommendations for people to contact
AndroidStudioProjects/MyFirstApplication/app/build/intermediates/res/merged/debug/values/values.xml:    <string name="app_name">My First Application</string>
Binary file AndroidStudioProjects/MyFirstApplication/app/build/intermediates/res/resources-debug.ap_ matches
AndroidStudioProjects/MyFirstApplication/app/src/main/res/values/strings.xml:    <string name="app_name">My First Application</string>
```

Reply:
mix fix doesn't works with project for multiple devices (wear, mobile etc)
```bash
→ grep "My Second Applicatiòñ" AndroidStudioProjects/* -R
AndroidStudioProjects/MySecondApplicatiòñ/.idea/workspace.xml:    <ignored path="My Second Applicatiòñ.iws" />
AndroidStudioProjects/MySecondApplicatiòñ/.idea/workspace.xml:    <favorites_list name="My Second Applicatiòñ" />
AndroidStudioProjects/MySecondApplicatiòñ/mobile/src/main/res/values/strings.xml:    <string name="app_name">My Second Applicatiòñ</string>
AndroidStudioProjects/MySecondApplicatiòñ/tv/src/main/res/values/strings.xml:    <string name="app_name">My Second Applicatiòñ</string>
AndroidStudioProjects/MySecondApplicatiòñ/wear/src/main/res/values/strings.xml:    <string name="app_name">My Second Applicatiòñ</string>
```

(yes I try wired chars too 😃 )

Reply:
did you have a file `.iws` in your projects?

Reply:
I see this:
```bash
grep "application-cafe-jeunesse-android" . -R | grep iws
./.idea/workspace.xml:    <ignored path="application-cafe-jeunesse-android.iws" />
```

but the file is not present:
```bash
❯ find . -name "*application-cafe-jeunesse-android.iws*"
```

Could it be the `favorites_list name`? 
```
.idea/workspace.xml:    <favorites_list name="application-cafe-jeunesse-android" />
```

`favorites_list` totally sounds like android-studio's project list.

Reply:
me too `.iws` doesn't exists

Favorites list can be renamed :-( 

I think this `<ignored path="My Second Applicatiòñ.iws" />` will be ok for most of the case. Can you confirm with your teammate?

For my information, did you have "old" projets - created with previous version of Android Studio?

Reply:
> For my information, did you have "old" projets - created with previous version of Android Studio?

Yes that's probably the case.

btw, I just tested something to find what the project name is:

<img width="474" alt="image" src="https://user-images.githubusercontent.com/1264761/31178670-c2b1c3b2-a8e7-11e7-8fd3-e9c5a02f0f08.png">

I renamed the folder and opened it again. From the list, it uses the directory name which holds the `.idea/workspace.xml`